src/Filter/Filter.BinaryComparison.js

```diff
@@ -0,0 +1,12 @@
+L.Filter.BinaryComparison = L.Filter.BinaryOperator.extend({
+  attributes: {
+    matchCase: false
+  },
+
+  initialize: function(firstValue, secondValue, matchCase) {
+    L.Filter.BinaryOperator.prototype.initialize.call(this, firstValue, secondValue);
+    if(matchCase !== undefined) {
```

Comment:
You can simply make this.attributes.matchCase = !!matchCase to avoid null and undefined values, also it helps to take only boolean values